scala/src/main/scala/com/scalableminds/webknossos/wrap/util/ExtendedTypes.scala

```diff
@@ -1,57 +1,23 @@
 package com.scalableminds.webknossos.wrap.util
 
-import java.io.RandomAccessFile
-import java.lang.reflect.Method
 import java.nio.MappedByteBuffer
 
 import net.liftweb.common.{Box, Failure, Full}
-import org.apache.commons.lang3.reflect.FieldUtils
 import net.liftweb.util.Helpers.tryo
 
 object ExtendedTypes {
 
-  implicit class ExtendedRandomAccessFile(f: RandomAccessFile) {
-    private val closedF = {
-      val method = f.getClass.getDeclaredField("closed")
-      method.setAccessible(true)
-      method
-    }
-
-    private val pathF = {
-      val method = f.getClass.getDeclaredField("path")
-      method.setAccessible(true)
-      method
-    }
-
-    def isClosed: Boolean = closedF.getBoolean(f)
-
-    def getPath: String = pathF.get(f).asInstanceOf[String]
-  }
-
   class ExtendedMappedByteBuffer(mappedData: MappedByteBuffer) {
-    val unsafe: AnyRef = FieldUtils.readField(mappedData, "unsafe", true)
-
-    val address: Long = FieldUtils.readField(mappedData, "address", true).asInstanceOf[Long]
-
-    val arrayBaseOffset: Long = FieldUtils.readField(mappedData, "arrayBaseOffset", true).asInstanceOf[Long]
-
-    val unsafeCopy: Method = {
-      val m = unsafe.getClass.getDeclaredMethod("copyMemory",
-        classOf[Object], classOf[Long], classOf[Object], classOf[Long], classOf[Long])
-      m.setAccessible(true)
-      m
-    }
 
     def capacity: Int = mappedData.capacity
 
     def copyTo(offset: Long, other: Array[Byte], destPos: Long, length: java.lang.Integer): Box[Unit] = {
       // Any regularly called log statements in here should be avoided as they drastically slow down this method.
       if (offset + length <= mappedData.limit()) {
         tryo {
-          val memOffset: java.lang.Long = address + offset
-          val targetOffset: java.lang.Long = destPos + arrayBaseOffset
-          // Anything that might go south here can result in a segmentation fault, so be careful!
-          unsafeCopy.invoke(unsafe, null, memOffset, other, targetOffset, length)
+          for (i <- 0 until length) {
+            other(destPos.toInt + i) = mappedData.get(offset.toInt + i)
```

Comment:
Note that the conversion to int effectively happened in the old version too. MappedByteBuffers only work with int. That’s why in WKWFile.scala a kind of paging algorithm is used that holds multiple such buffers for larger files.

Reply:
I see. However, I would prefer to change the signature of `destPos` and `offset` to int to reflect that limitation. If you don't want to change the signature for compatibility reasons, I would still prefer to do the casting before the loop.